### ISSUE TITLE
Error with search query starting with "or"

### ISSUE THREAD

**Issue by dinojr** [xuchunyang/helm-notmuch#8]
The message is:
`error in process filter: Search failed: "["`

**Comment by xuchunyang** [comment]
Unfortunately, I stopped using Notmuch long time ago and I am afraid I won't investigate the issue anytime soon. But meanwhile helm-notmuch is a very simple package - it uses the notmuch command line tool to collect search results and display them using helm, so if the issue is caused by helm-notmuch, it should not be very hard to figure out.

**Comment by dinojr** [comment]
I'll give it a try then.

**Comment by Ambrevar** [comment]
@xuchunyang: Out of curiosity, what are you using now and why did you stop using Notmuch?

**Comment by xuchunyang** [comment]
@Ambrevar I'm not a heavy user of email, e.g., I sent only one email since this month.

**Comment by Ambrevar** [comment]
I think I've got and answer to this:

```diff
(defun helm-notmuch-candidate-formatter (cand)
  "Format the single entry CAND."
+  (if (not (string-match-p "\\[" cand ))
+      ""
  ...
```

Basically when querying something that Xapian does not understand, notmuch returns an error.  Example
``` shell
> notmuch search "date:foo"
notmuch search: A Xapian exception occurred
A Xapian exception occurred parsing query: Didn't understand date specification 'foo'
Query string was: date:foo
```

In `helm-notmuch-collect-candidates`, instead of `start-process` we could use something like this:

```
(make-process :name "helm-notmuch"
              :buffer helm-buffer
              :command cmds
              :connection-type 'pipe
              :stderr helm-notmuch-log-buffer)
```

**Comment by Ambrevar** [comment]
Note that the first suggestion is the real fix, the second is only an improvement in terms of error reporting.

@xuchunyang: Would you take a PR?

**Comment by xuchunyang** [comment]
@Ambrevar Yes, I'll accept such PR as long as it improves the situation.

**Comment by Ambrevar** [comment]
Fixed with dae426c68635c7f586bcc070f68cefbbfa802f20.
